APMrover2/Parameters.cpp

```diff
@@ -603,6 +603,24 @@ const AP_Param::GroupInfo ParametersG2::var_info[] = {
     // @Path: ../libraries/AC_Avoidance/AP_OAPathPlanner.cpp
     AP_SUBGROUPINFO(oa, "OA_", 45, ParametersG2, AP_OAPathPlanner),
 
+    // @Param: MAX_SPEED
+    // @DisplayName: Maximum vehicle speed
+    // @Description: Maximum speed vehicle can obtain at full throttle. If 0, it will be estimated based on CRUISE_SPEED and CRUISE_THROTTLE.
+    // @Units: m/s
+    // @Range: 0 100
+    // @Increment: 0.1
+    // @User: Advanced
+    AP_GROUPINFO("MAX_SPEED", 46, ParametersG2, max_speed, 0.0f),
```

Comment:
SPEED_MAX would be better than MAX_SPEED.  Another alternative is PILOT_SPEED_MAX.

One way or another I think users are going to be confused about this parameter vs WP_SPEED.  I suspect that users will expect that SPEED_MAX applies to all modes.

In Copter and (to a lesser extent Rover) we prepend PILOT_ for parameters that apply to pilot input or manual modes.

Sorry, I don't immediately have a good answer but I worry users are going to be confused but that doesn't mean we have to hold up the PR.

Reply:
.. I suspect that we should call it SPEED_MAX and then use it from now on as the maximum speed of the vehicle.  So at some point we should add a check that WP_SPEED is always less than SPEED_MAX.

Reply:
change made